buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/BuildForkPointDistribution.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing;
+
+import org.apache.commons.io.FileUtils;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.CacheableTask;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.OutputDirectory;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.stream.Stream;
+
+@CacheableTask
+public class BuildForkPointDistribution extends DefaultTask {
```

Comment:
2min 4s, see here: https://builds.gradle.org/viewLog.html?buildId=16580656&buildTypeId=Gradle_Check_PerformanceTestCoordinator&tab=buildLog&_focus=3176#_state=69,3176

Another option would be query from Coordinator artifacts via TC, but I think it's a bit more complicated.

Reply:
And it's not only for a single branch - the `BuildForkPointDistribution`'s input is only commit id, so therotically all branched forked from the same commit will have cache hit - but given that we have lots of commits everyday, this might not that beneficial.

Reply:
Looks like creating the Kotlin DSL accessors is where most of the time is spent. See the gap in the timeline: https://e.grdev.net/s/cbxf7vlqfckwm/timeline?outcomeFilter=FROM-CACHE&sort=duration

I guess that will be much faster soon, right @bamboo?

Reply:
I'm not sure I understand this gap... what happened in this gap?

Reply:
My guess would be generating the Kotlin DSL accessor jars.